app/models/service_retire_task.rb

```diff
@@ -42,22 +42,22 @@ def create_retire_subtasks(parent_service, parent_task)
       # Initial Options[:dialog] to an empty hash so we do not pass down dialog values to child services tasks
       nh['options'][:dialog] = {}
       new_task = create_task(svc_rsc, parent_service, nh, parent_task)
-      create_retire_subtasks(svc_rsc.resource, new_task) if svc_rsc.resource.kind_of?(Service)
       new_task.after_request_task_create
       miq_request.miq_request_tasks << new_task
       new_task.tap(&:deliver_to_automate)
     end.compact!
   end
 
   def create_task(svc_rsc, parent_service, nh, parent_task)
-    (svc_rsc.resource.type.demodulize + "RetireTask").constantize.new(nh).tap do |task|
+    resource_type = svc_rsc.resource.type.presence || "Service"
+    (resource_type.demodulize + "RetireTask").constantize.new(nh).tap do |task|
       task.options.merge!(
-        :src_id              => svc_rsc.resource.id,
+        :src_ids             => svc_rsc.resource.id,
```

Comment:
Should this be an array since `src_ids` is plural?

Reply:
At this point, no, because it's the naming that's the problem. This will only ever be one thing at this point in the code. It's still called with the plural name because the UI allows us to retire either an object by ```src_id``` or ```src_ids``` but by the time we get to task creation we've accounted for that. I just haven't gotten around to changing the name.

Reply:
@d-m-u Are you going to change this to an array, so its consistent with the other call. Also the Active Record find works with a single id or an array of ids

Reply:
@mkanoor we discussed yesterday the fact that there's only going to be one resource at this point. I know the naming is bad. It's been bad since the initial commit five years ago. Can I fix it later?

Reply:
I'm just going to have to fix this later anyway. It shouldn't be an array, no.